bin/utils.js

```diff
@@ -228,16 +242,44 @@ const closeConn = (doc, conn) => {
     doc.conns.delete(conn)
     awarenessProtocol.removeAwarenessStates(doc.awareness, Array.from(controlledIds), null)
     if (doc.conns.size === 0 && persistence !== null) {
-      // if persisted, we store state and destroy ydocument
-      persistence.writeState(doc.name, doc).then(() => {
-        doc.destroy()
-      })
-      docs.delete(doc.name)
+      cancelClosing(doc.name)
+      closingTimeouts.set(
+        doc.name,
+        setTimeout(
+          () => {
+            closingTimeouts.delete(doc.name)
+            if (doc.conns.size === 0) {
+              // if persisted, we store state and destroy ydocument
+              persistence.writeState(doc.name, doc)
+                .then(
+                  async () => {
+                    if (doc.conns.size === 0) {
+                      docs.delete(doc.name)
+                      doc.destroy()
+                    }
+                  }
+                )
+            }
+          },
+          WRITE_STATE_DEBOUNCE_WAIT
+        )
+      )
     }
   }
   conn.close()
 }
 
+/**
+ * @param {string} docName
+ */
+export const cancelClosing = (docName) => {
```

Comment:
no way to call this within this package?

Reply:
no we do call it within this package. I just exported it so we can call it outside too if we want

Reply:
meh i'll remove it.